### PR TITLE
Update PyPy and FOSS EDA tools

### PR DESCRIPTION
**Pull request by emilazy**
###### Motivation for this change

I initially started by wanting to just bump `yosys`/`symbiyosys`/`icestorm`/`trellis`/`nextpnr`, but ran into `icestorm` and `nextpnr` being broken due to PyPy being marked as broken. After discussing on IRC with @thoughtpolice and @MostAwesomeDude, I decided to update PyPy and disable its tests for now, as a bunch of tests are already being disabled and it's a significant speed-up for `icestorm` (as documented in 18839e1). `pypy35` becomes `pypy36`, since upstream changed the py3 version they target and I assume nobody particularly wants to use an old beta version of `pypy35`.

The newer version of yosys is required for e.g. [nmigen](https://github.com/m-labs/nmigen).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/icestorm/default.nix [hunk 1]
```diff
@@ -1,23 +1,21 @@
 { stdenv, fetchFromGitHub
 , pkgconfig, libftdi
 , python3, pypy3
+, usePyPy ? stdenv.isx86_64 /* pypy3 seems broken on i686 */
```
Comment (FRidh) [inline pkgs/development/tools/icestorm/default.nix]: please document here the motivation for using pypy as a default, given that is quite uncommon.

**Review by FRidh — APPROVED** [review]

File: pkgs/development/tools/icestorm/default.nix (hunk 1 above)
Comment (emilazy) [reply pkgs/development/tools/icestorm/default.nix]: ok, I added a comment

**Comment by thoughtpolice** [comment]
@GrahamcOfBorg build icestorm trellis nextpnr yosys symbiyosys

**Review by thoughtpolice — APPROVED** [review]

**Comment by thoughtpolice** [comment]
@emilazy Thank you for doing all this! Let's see how borg responds; in particular, `-DSERIALIZE_CHIPDB=OFF` for NextPNR may not be appropriate for the Hydra builders in the long run if it uses *too* much memory, so we might have to remove it (should that happen - we'll cross that bridge when we get to it.)

**Comment by emilazy** [comment]
FWIW, `-DSERIALIZE_CHIPDB=OFF` was (effectively) default [until recently](https://github.com/YosysHQ/nextpnr/issues/296). The serialization slows down the build a lot and I personally don't have any issues with building it on a 16 GiB RAM laptop, though I do have swap.

Not sure what happened with the aarch64-linux build there...

**Comment by thoughtpolice** [comment]
Great to know. The aarch64 failure is anomalous, but it looks like x86_64 tried to build pypy 7.1.1 from source since it's not using the prebuilt expression. This causes the build to timeout since pypy takes A Long Time. I suppose what we actually want to do is perhaps split the PyPy update into its own PR, maybe.

I'll test this on my own machine later today.

**Comment by andersk** [comment]
pypy36_prebuilt doesn't build:

```
checking for references to /build/ in /nix/store/00h1ivkm5rvaq9cryqynvsrccx8ibi38-pypy3_prebuilt-7.1.1...
running install tests
Testing whether we can import modules
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/nix/store/00h1ivkm5rvaq9cryqynvsrccx8ibi38-pypy3_prebuilt-7.1.1/lib-python/3/ssl.py", line 100, in <module>
    import _ssl             # if we can't import it, let the error propagate
  File "/nix/store/00h1ivkm5rvaq9cryqynvsrccx8ibi38-pypy3_prebuilt-7.1.1/lib_pypy/_ssl/__init__.py", line 1, in <module>
    from _cffi_ssl._stdssl import (_PROTOCOL_NAMES, _OPENSSL_API_VERSION,
  File "/nix/store/00h1ivkm5rvaq9cryqynvsrccx8ibi38-pypy3_prebuilt-7.1.1/lib_pypy/_cffi_ssl/_stdssl/__init__.py", line 6, in <module>
    from _pypy_openssl import ffi
ImportError: libssl.so.1.0.0: cannot open shared object file: No such file or directory
builder for '/nix/store/0mkirnflycy1njynrfkz0f7snqn1xm81-pypy3_prebuilt-7.1.1.drv' failed with exit code 1
```

**Comment by FRidh** [comment]
I bumped a fix to master, b75f9a5f6de557e1035ece624909569a3c1b6d34

**Pull request merged**